src/app/static/src/app/components/genericChart/GenericChart.vue

```diff
@@ -102,34 +142,105 @@
             },
             chartConfigValues() {
                 const dataSourceConfigValues = this.chartMetadata.dataSelectors.dataSources.map((dataSourceConfig) => {
+                    const selections = this.dataSourceSelections[dataSourceConfig.id];
                     return {
-                        selections: this.dataSourceSelections[dataSourceConfig.id],
+                        selections,
                         editable: dataSourceConfig.type === "editable",
-                        config: dataSourceConfig
+                        config: dataSourceConfig,
+                        filters: this.datasets[selections.datasetId]?.metadata.filters
                     }
                 });
+
+                //Provide additional metadata to jsonata relating to subplots (rows and columns)
+                //and define scroll height
+                const layoutData = {} as Dict<unknown>;
+                let scrollHeight = "100%";
+                const subplots = this.chartMetadata.subplots;
+                if (subplots && this.chartData) {
+                    const distinctAreas = new Set(this.chartData["data"].map((row: any) => row[subplots.distinctColumn]));
+                    const numberOfPlots = [...distinctAreas].length;
```

Comment:
This requires the change in ts.config (`downlevelIteration`) to allow iteration of the Set.